intelligent_placer_lib/detection.py

```diff
@@ -1,101 +1,132 @@
+    poly_index = np.argmin([int(cv2.moments(cnt)['m10'] / cv2.moments(cnt)['m00']) for cnt in good_cnts])
     if verbose:
+        im = image.copy()
+        cv2.drawContours(im, [good_cnts[poly_index]], -1, (0, 0, 255), 5)
         plt.imshow(im)
-        plt.title('image contours')
+        plt.title('polygon contour')
         plt.show()
-    return list(filter(lambda contour: cv2.contourArea(contour) >= MIN_OBJECT_AREA, contours))
-
-
-def _get_transformed_mask(segmented: np.ndarray, src_pts: np.ndarray, dst_pts: np.ndarray, source_mask: np.ndarray,
-                          object_class: int) -> Optional[np.ndarray]:
-    src_key_pts = np.float32([src_pts]).reshape(-1, 1, 2)
-    dst_key_pts = np.float32(dst_pts).reshape(-1, 1, 2)
-
-    matrix, mask = cv2.findHomography(src_key_pts, dst_key_pts, cv2.RANSAC, 5.0)
-    if matrix is None:
-        return None
-
-    p = np.where(source_mask >= 200)
-    mask_points = [[point[1], point[0]] for point in zip(*p)]
-    new_mask_points = cv2.perspectiveTransform(np.float32(mask_points).reshape(-1, 1, 2), matrix)[:, 0, :]
-    for point in new_mask_points:
-        if segmented.shape[0] <= int(point[1]) or int(point[1]) < 0 \
-                or segmented.shape[1] <= int(point[0]) or int(point[0]) < 0:
-            return None
-        if segmented[int(point[1])][int(point[0])] != 0 and segmented[int(point[1])][int(point[0])] != object_class:
-            return None
-        segmented[int(point[1])][int(point[0])] = object_class
-    plt.imshow(segmented + 230)
-    plt.show()
-    print(new_mask_points)
-    return segmented
+    return good_cnts[poly_index], good_cnts[:poly_index]+good_cnts[poly_index+1:]
+
+
+def _get_contour_mask(contour, shape):
+    mask = np.zeros(shape)
+    return cv2.drawContours(mask, [contour], -1, 255, cv2.FILLED)
+
+
+def get_channels_average(object_mask: np.ndarray, object_image: np.ndarray) -> np.ndarray:
+    object_part = np.array([object_image[..., i] * (object_mask / 255).astype('uint8') for i in range(object_image.shape[-1])])
+    return np.sum(np.sum(object_part, axis=1), axis=1) / np.sum(object_mask > 0) / 255
+
 
+def is_red_circle(object_mask: np.ndarray, object_image: np.ndarray, object_contour, circle_contour) -> bool:
+    channels_av = get_channels_average(object_mask, object_image)
+    is_red = not np.isnan(channels_av[0]) and channels_av[0] > 0.6 and channels_av[1] < 0.3 and channels_av[2] < 0.3
+    if not is_red:
+        return False
+    cur_res = cv2.matchShapes(object_contour, circle_contour, cv2.cv2.CONTOURS_MATCH_I3, 0.0)
+    return cur_res < 0.1
 
-def _classify_objects(contours, image: np.ndarray, verbose: bool = False) -> Tuple[np.ndarray, List]:
+
+def is_blue_rectangle(object_mask: np.ndarray, object_image: np.ndarray, object_contour) -> bool:
+    channels_av = get_channels_average(object_mask, object_image)
+    is_blue = not np.isnan(channels_av[2]) and channels_av[2] > 0.6 and channels_av[0] < 0.2 and channels_av[1] < 0.6
```

Comment:
Should these constants be in the code? Maybe create variables for them. 
Check the same in all methods